### DISCUSSION TITLE
option key misbehaves with zellij

### DISCUSSION THREAD

**Discussion by pdeva** [ghostty-org/ghostty#3207] [Q&A] [answered]
this happens regardless of whether i set `macos-option-as-alt` to either `true` or `false`. 

i am on macos.

i use ghostty to ssh into a linux box and run `zellij` there.

on all other terminals, i am able to use uses option+ left/right arrow key to navigate between zellij panes.

on ghostty it doesnt work.

**Comment by qwerasd205** [comment, marked as answer]
On macOS we by default bind `alt+left` to `esc:b` and `alt+right` to `esc:f` since a lot of macOS users expect this behavior (despite it being non-standard) because Terminal.app does it. You can unbind them with `keybind = alt+left=unbind` and `keybind = alt+right=unbind`, then it may work.

**Reply by pdeva** [reply]
works with this!

**Reply by dj95** [reply]
Can also confirm, that this fixes the issue for me.

**Reply by currenthandle** [reply]
https://github.com/ghostty-org/ghostty/discussions/3207#discussioncomment-11713277

**Reply by droidsam** [reply]
In case this option doesn't help, check [here](https://github.com/ghostty-org/ghostty/discussions/3207#discussioncomment-14186031)

**Comment by mitchellh** [comment]
If you're over SSH, this can also be a terminfo issue: https://ghostty.org/docs/help/terminfo

**Comment by qballer** [comment]
These are my current configuration 
```
keybind = alt+left=unbind
keybind = alt+right=unbind
macos-option-as-alt = true
```
And it doesn't seem to solve the issue, any ideas on whats the issue here?

**Reply by currenthandle** [reply]
In my situation I'm not using zelij. I am on a mac, seeing the issue in and outside of tmux.

this fixed it for me:
```
keybind = alt+left=unbind
keybind = alt+right=unbind
```

with or without 

```
macos-option-as-alt = true
```

should I keep 

```
macos-option-as-alt = true
```

for some reason?

**Reply by Staindk** [reply]
TL;DR if you end up here with `Option` issues in base ghostty / tmux (unsure about zellij), check your configs and remove old fixes that might now be breaking things.

For anyone that struggles with `Option + Left` and `Option + Right` who ends up here and finds these answers don't help, I assume something changed in ghostty or MacOS at some point and these fixes are no longer needed/relevant.

My config had this in it (from around 6 months ago):
```
keybind = option+left=unbind
keybind = option+right=unbind
```

... I think I set that to fix something, but at some point it caused my option + left/right to break (instead of moving word-by-word it would just print `3D;` and `3C;` codes).

Trying the fixes in this discussion didn't help, but removing all my `option` keybinds fixed everything for me.

**Reply by droidsam** [reply]
I've been struggling with this issue since April 2025, and none of these fixes have helped. I checked my profile, and I don't have any Option bindkey set.

Anything I can try to debug this?

**Reply by droidsam** [reply]
Update: I tried these configs, and it works for me on MacOS 15.6

Shell Configuration (e.g., Zsh):
If the issue persists, you might need to explicitly bind the Option-arrow keys in your shell's configuration file (e.g., .zshrc for Zsh).
Add the following lines to your .zshrc file:

```
        bindkey "^[[1;3D" backward-word  # Option + Left
        bindkey "^[[1;3C" forward-word   # Option + Right
```
After modifying your .zshrc, source it or restart your shell for the changes to take effect.

Ghostty Config

```
        keybind = alt+left=unbind
        keybind = alt+right=unbind
```
And with that Opt + Arrows works on MacOS

**Comment by ben-albrecht** [comment]
Has anyone been able to get alt-f working within tmux in ghostty on macos? I can only get this working in iterm2 (see [this post](https://superuser.com/questions/649960/option-key-does-not-work-as-meta-in-tmux)).

**Reply by pluiedev** [reply]
Please don't post in already resolved discussions — make a new one instead
